modules/omsBidAdapter.js

```diff
@@ -0,0 +1,321 @@
+        w: screen.width,
+        h: screen.height
+      },
+      tmax: bidderRequest?.timeout
+    };
+
+    if (bidderRequest?.gdprConsent) {
+      deepSetValue(payload, 'regs.ext.gdpr', +bidderRequest.gdprConsent.gdprApplies);
+      deepSetValue(payload, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+    }
+
+    if (bidderRequest?.uspConsent) {
+      deepSetValue(payload, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+    }
+
+    if (config.getConfig('coppa') === true) {
+      deepSetValue(payload, 'regs.coppa', 1);
+    }
+
+    if (bidReqs?.[0]?.schain) {
+      deepSetValue(payload, 'source.ext.schain', bidReqs[0].schain)
+    }
+
+    if (bidReqs?.[0]?.userIdAsEids) {
+      deepSetValue(payload, 'user.ext.eids', bidReqs[0].userIdAsEids || [])
+    }
+
+    if (bidReqs?.[0].userId) {
+      deepSetValue(payload, 'user.ext.ids', bidReqs[0].userId || [])
+    }
+
+    return {
+      method: 'POST',
+      url: URL,
+      data: JSON.stringify(payload),
+    };
+  } catch (e) {
+    logError(e, {bidReqs, bidderRequest});
+  }
+}
+
+function isBidRequestValid(bid) {
+  if (bid.bidder !== BIDDER_CODE || !bid.params || !bid.params.publisherId) {
+    return false;
+  }
+
+  return true;
+}
+
+function interpretResponse(serverResponse) {
+  let response = [];
+  if (!serverResponse.body || typeof serverResponse.body != 'object') {
+    logWarn('OMS server returned empty/non-json response: ' + JSON.stringify(serverResponse.body));
+    return response;
+  }
+
+  const {body: {id, seatbid}} = serverResponse;
+
+  try {
+    if (id && seatbid && seatbid.length > 0 && seatbid[0].bid && seatbid[0].bid.length > 0) {
+      response = seatbid[0].bid.map(bid => {
+        return {
+          requestId: bid.impid,
+          cpm: parseFloat(bid.price),
+          width: parseInt(bid.w),
+          height: parseInt(bid.h),
+          creativeId: bid.crid || bid.id,
+          currency: 'USD',
+          netRevenue: true,
+          mediaType: BANNER,
+          ad: _getAdMarkup(bid),
+          ttl: 60,
+          meta: {
+            advertiserDomains: bid?.adomain || []
+          }
+        };
+      });
+    }
+  } catch (e) {
+    logError(e, {id, seatbid});
+  }
+
+  return response;
+}
+
+// Don't do user sync for now
+function getUserSyncs(syncOptions, responses, gdprConsent) {
+  return [];
+}
+
+function onTimeout(timeoutData) {
+  if (timeoutData === null) {
+    return;
+  }
+
+  _trackEvent('timeout', timeoutData);
+}
+
+function onBidderError(errorData) {
+  if (errorData === null || !errorData.bidderRequest) {
+    return;
+  }
+
+  _trackEvent('error', errorData.bidderRequest)
+}
+
+function onBidWon(bid) {
+  if (bid === null) {
+    return;
+  }
+
+  _trackEvent('bidwon', bid)
+}
+
+function _trackEvent(endpoint, data) {
+  ajax(`${TRACK_EVENT_URL}/${endpoint}`, null, JSON.stringify(data), {
+    method: 'POST',
+    withCredentials: false
+  });
+}
+
+function _isMobile() {
```

Comment:
User agents are going away, please also inspect the device.sua object on the request

Reply:
The `device.sua` object doesn't effectively determine the device type according to our requirements. Here's an example:

For identifying mobile devices, I utilized the user agent associated with the device:

`Mozilla/5.0 (Linux; Android 10; SM-G960F) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.5845.115 Mobile Safari/537.36` 

The resulting sua object would appear as follows:

![image](https://github.com/prebid/Prebid.js/assets/3617478/4c8a6742-8dfc-4c6b-a1eb-3a626786a669)

This logic is similarly applied for Apple TV:

`AppleCoreMedia/1.0.0.21K646 (Apple TV; U; CPU OS 17_3 like Mac OS X; pt_pt)` 

![image](https://github.com/prebid/Prebid.js/assets/3617478/07fbf582-c1fc-4f40-8292-cf0beabf14c9)

I understand that the entire logic is based on headers. The only update I propose is to check the `device.sua.mobile ` field in the same line where I check the user agent, as shown below:

``` javascript
 if (sua?.mobile || (/(ios|ipod|ipad|iphone|android)/i).test(ua)) {
    return 1
  }
  ```